src/Data/Blob.hs

```diff
@@ -31,69 +27,52 @@ module Data.Blob
 
 import Prologue
 
+import           Analysis.File (fileBody)
+import qualified Analysis.File
+import           Analysis.Language as Language
 import           Control.Effect.Error
 import           Data.Aeson
 import qualified Data.ByteString.Lazy as BL
 import           Data.Edit
 import           Data.JSON.Fields
-import           Data.Language
 import           Data.Module
-import           Source.Source (Source)
+import           Source.Source (Source, totalSpan)
 import qualified Source.Source as Source
 import qualified System.FilePath as FP
 import qualified System.Path as Path
 import qualified System.Path.PartClass as Path.PartClass
 
--- | A 'FilePath' paired with its corresponding 'Language'.
--- Unpacked to have the same size overhead as (FilePath, Language).
-data File = File
-  { filePath     :: FilePath
-  , fileLanguage :: Language
-  } deriving (Show, Eq)
```

Comment:
The `File` type in `semantic-analysis` does not hold a `Language` member. However, it has an extra field, used in other places to hold parsed terms; we can stick a `Language` in there, thereby yielding a type equivalent in power to the old `File`.

Reply:
We might consider it worth adding a language field to `Analysis.File.File`, but I’ll reserve judgement till we see how it feels.